pkg/apiclients/fileupload/client.go

```diff
@@ -36,9 +35,10 @@ type HTTPClient struct {
 
 // Client defines the interface for the high level file upload client.
 type Client interface {
-	CreateRevisionFromPaths(ctx context.Context, paths []string) (UploadResult, error)
+	CreateRevisionFromPaths(ctx context.Context, paths []string, rootPath string) (UploadResult, error)
```

Comment:
@alina-d-m quick thought/question, will you be using any of these other Create functions? Or will you be working with `CreateRevisionFromChan()`?
If you will be using CreateRevisionFromChan(), I would propose to remove these convenience functions actually. 
Everything can be achieved through CreateRevisionFromChan and it is not clear if any of them is actually helpful/required.

Reply:
Okay, this sounds awesome. Thank you for making them work! I only left them there because of the missing channel version, so now that you have it, I think we can make this implementation even more focussed and slim, which will make it easier to test and maintain.

Sorry that this approach created some extra iterations and work. But I think `CreateRevisionFromChan()` is the perfect minimal interface and we should encourage its usage!

Reply:
The migration will not work 1:1 anyways, since we will exclude filtering from the fileupload API. This means, they will use channel based API anyways.